### PR TITLE
Alexander Gavruk. 09-redux

### PR DESCRIPTION
**Pull request by KuzyaCat**

### CODE REVIEW

File: 09-redux/classworks/Gavruk_Alexander/src/components/SearchResults/SearchResults.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.search-results {
```
Comment (mikitasolo) [inline 09-redux/classworks/Gavruk_Alexander/src/components/SearchResults/SearchResults.css:1]: cool class!

File: 09-redux/classworks/Gavruk_Alexander/src/pages/Routes.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Switch, Route, Redirect } from 'react-router-dom';
+
+import useRedditApi from 'hooks/useRedditApi';
+
+import Login from 'pages/login';
+import Callback from 'pages/callback';
+import Home from 'pages/home';
+
+function Routes(props) {
```
Comment (mikitasolo) [inline 09-redux/classworks/Gavruk_Alexander/src/pages/Routes.js]: you don't need to get props here and pass farther anymore as you use Redux

File: 09-redux/classworks/Gavruk_Alexander/src/resources/reducer.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { combineReducers } from 'redux';
+
+import subreddit from './subreddit/subreddit.reducer';
+
+const reducers = {
+  subreddit,
+};
+
+const rootReducer = combineReducers({
+  ...reducers,
```
Comment (mikitasolo) [inline 09-redux/classworks/Gavruk_Alexander/src/resources/reducer.js]: why do you create a new object?

File: 09-redux/classworks/Gavruk_Alexander/src/pages/home/Home.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  const isPostsData = useSelector(subredditSelectors.getIsPostsData);
+  const searchValue = useSelector(subredditSelectors.getSearchValue);
+  const communityTitleData = useSelector(subredditSelectors.getCommunityTitleData);
+  const subredditData = useSelector(subredditSelectors.getSubredditData);
+
+  async function fetchData() {
+    const { fetchReddit } = props;
+    const url = (searchValue === '' || isPostsData) ? `/${communityTitleData.communityUrl}/hot` : `/search?q=${searchValue}&type=sr,user`;
+    const data = await fetchReddit(url).then(res => res.json());
```
Comment (mikitasolo) [inline 09-redux/classworks/Gavruk_Alexander/src/pages/home/Home.js]: this logic should not be in component. it should be in action

**Review by mikitasolo — CHANGES_REQUESTED** [review]

File: 09-redux/classworks/Gavruk_Alexander/src/components/SearchResults/SearchResults.css (hunk 1 above)
Comment (KuzyaCat) [reply 09-redux/classworks/Gavruk_Alexander/src/components/SearchResults/SearchResults.css:1]: @SobakaSlava ty

File: 09-redux/classworks/Gavruk_Alexander/src/pages/Routes.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import Callback from 'pages/callback';
+import Home from 'pages/home';
+
+function Routes() {
+  const [,,isLoggedIn] = useRedditApi();
+
+  if (isLoggedIn) { 
+    return (
+      <Switch>
+        <Route path="/" exact component={() => <Home />} />
```
Comment (mikitasolo) [inline 09-redux/classworks/Gavruk_Alexander/src/pages/Routes.js:16]: as we don't pass anything anymore we can make just `component={Home}`. but let's leave it as it is here

**Review by mikitasolo — APPROVED** [review]

**Pull request merged**